src/libmscore/sym.h

```diff
@@ -21,13 +21,30 @@
 #include "ft2build.h"
 #include FT_FREETYPE_H
 
+// Needs to be duplicated here and in style.h since moc doesn't handle macros from #include'd files
+#ifdef SCRIPT_INTERFACE
+#define BEGIN_QT_REGISTERED_ENUM(Name) \
```

Comment:
It might be better to declare these definitions in one place (separate h file) and use them where necessary instead of duplicating them.

Reply:
I have tried including it from a separate file when working on #6205 but that didn't work for me. Therefore I have duplicated the macro and add the comment (two lines above) explaining that by `moc` not handling included files.

However I have just checked it again, and it turns out that I was wrong: `moc` does process included files, and the real issue here is just that `moc` doesn't get invoked for `style.h` and `sym.h` if this macro is moved out of these files. The reason is that `moc` invocations are handled by CMake automoc mechanism which detects the files to be processed by certain criteria including presence of macros like `Q_OBJECT` and `Q_GADGET`. As `style.h` and `sym.h` don't contain it except for this macro, automoc doesn't launch moc on those files if the macro is moved out of them.

With CMake versions newer than 3.10 this can be fixed by listing the name of this macro in [AUTOMOC_MACRO_NAMES](https://cmake.org/cmake/help/latest/prop_tgt/AUTOMOC_MACRO_NAMES.html) property. This indeed works but this solution seems to be applicable only for master branch (which currently requires CMake 3.16) but not for 3.x (which sets the minimum CMake version to 3.3). I have added a commit with `AUTOMOC_MACRO_NAMES` to this PR so these definitions are no longer duplicated here.

With older versions of CMake launching moc on these files seems to require either creating dummy classes with macros recognized by automoc (like it is done in [`types.h`](https://github.com/musescore/MuseScore/blob/1304cfd1309ed98c9516a8a5697ab80f19c699af/libmscore/types.h#L630-L634)) or manually listing the relevant headers in [`qt5-wrap-cpp`](https://doc.qt.io/qt-5/qtcore-cmake-qt5-wrap-cpp.html) macro, so I guess for the current 3.x branch there is no solution that would be much better than duplicating this macro.